### PR TITLE
fix: set TrackerDB blocked by default flag to `true` & refactor exceptions blocking rules

### PR DESCRIPTION
**Pull request by smalluban**
### Overview

* Sets all TrackerDB categories to be blocked by default
* Fixes issue with the state where the tracker is allowed for all sites and blocked for a specific domain
* Technically, it cleans out the `/background/exceptions.js` code, as the cosmetic filters were never supported, and the whole code (except the exceptions storage) only applies to Chromium-based browsers

In general, this PR makes even the logic in Firefox and Chromium-based browsers. The TrackerDB is no longer used to block requests. It is only used to detect or allow the tracker (globally or on the domain level).

### Testing focus

Open google.pl, try to allow "Google" tracker for all sites, and block the "google.pl" domain. On `main` the tracker is not blocked again. Expected result: "Google" requests are allowed on any google domain except the `google.pl`. 

The PR touches the exceptions DNR rules, so it might be worth doing some smoke tests of all 4 states of the exceptions. I did some testing on Safari, and it looks promising there. We can remember to check out it again for Safari on release.

FYI @chrmod : This PR removes the need for TrackerDB  DNR ruleset for essentials. After the merge, we can stop creating it.

### CODE REVIEW

**Comment by smalluban** [comment]
@GRadziejewski Please wait with testing until technical review is done and PR is approved (Make it a default behavior for upcoming PRs).

File: extension-manifest-v3/scripts/download-engines/index.js
Comment (chrmod) [inline extension-manifest-v3/scripts/download-engines/index.js:27]: what's the reason to avoid console.log and use `process.stdout`?

Comment (smalluban) [reply extension-manifest-v3/scripts/download-engines/index.js:27]: Only then you can write in "the same line", the `console.log` adds `\n` as default. This is UI fix for logging downloaded engines - if they are fetched, then they are displayed one by one in lines, but if they are cached, they are displayed only during the check, and then next engine replaces the line. This is common pattern for building logs, to not overblow the console.

File: extension-manifest-v3/src/background/exceptions.js
Comment (smalluban) [reply extension-manifest-v3/src/background/exceptions.js:64]: Ok

File: extension-manifest-v3/src/background/exceptions.js
Comment (smalluban) [reply extension-manifest-v3/src/background/exceptions.js:38]: Look at the implementation, those functions return a function, which actually creates a document of offscreen

File: extension-manifest-v3/scripts/download-engines/index.js
Comment (smalluban) [reply extension-manifest-v3/scripts/download-engines/index.js:65]: Yes, as it might help when logs could help, from which CDN the lists were fetched.

**Review by chrmod — APPROVED** [review]

**Comment by smalluban** [comment]
@GRadziejewski You are free to test. You can find everything needed in the main description.

**Comment by GRadziejewski** [comment]
## Smoke tests Iteration no 1:
Tests of: https://github.com/ghostery/ghostery-extension/pull/1850#issuecomment-2320727816

---
Checked features:

- Enable/Disable Ad-blocking 
- Enable/Disable Anti-tracking			
  - Trusted 0 categories
  - Trust / block the tracker
  - URL / Trackers blocked on UI
- Enable/Disable Never Consent  	
  - Check website with - CPM Onetrust ([booking.com](https://www.ghostery.com/whotracksme/websites/booking.com)) 
  - Check website with - CPM Evidon ([playstation.com](https://www.ghostery.com/whotracksme/websites/playstation.com ))  									
  - Check website with - CPM Airbnb ([airbnb.com](https://www.ghostery.com/whotracksme/websites/airbnb.com))  											
- Pause Ghostery
- Log into Google account and open google.com 	
  - Check YT

```
Browser: Chrome 128.0.6613.114
OS: Windows 10
GBE 10.4.3

Browser: Firefox 129.0.2
OS: Windows 11
GBE 10.4.3
```
---


## Issues:
- 

## Fixed:
- 

## Cannot reproduce:
- 

## Will not be fixed:
-

**Review by GRadziejewski — APPROVED** [review]
LGTM

**Pull request merged**
